src/Contexts/section.actions.jsx

```diff
@@ -0,0 +1,417 @@
+ * 
+ * @param {Object} [section] - section object, all necesarry data are setten by default and can be overwirten
+ */
+export const addSection = (section) => ({
+	type: ADD_SECTION,
+	payload: {
+		title: "",
+		_id: uuid(), // TODO: mark, so it will not be saved to backend with this id
+		visible: true,
+		docValue: {},
+		changed: new Set(),
+		position: -1, // added at last pos
+		...section
+	}
+})
+
+/**
+ * Fetch one or multible sections from the server and add it to sections state
+ * if no positon is setted by the server response it will be added at last position
+ *
+ * @param  {...String} sectionIds - comma seperated list of section ids
+ */
+export const fetchSection = (...sectionIds) => ({state, dispatch}) => {
+
+	// TODO: check connection and inform user
+
+	const proms = sectionIds.map(async (_id) => {
+
+		dispatch({
+			type: FETCHING_SECTION,
+			payload: _id
+		})
+
+		return editorWS.emit(
+			'get',
+			`lesson/${state.lesson._id}/sections`,
+			_id
+		)
+	})
+
+	const resolved = Promise.allSettled(proms)
+	resolved.forEach((res, i) => {
+		if(res.status === 'fulfilled'){
+			dispatch(addSection(res.value))
+			dispatch({
+				type: SECTION_FETCHED,
+				payload: res.value._id
+			})
+		} else {
+			dispatch({
+				type: FETCHING_SECTION_FAILED,
+				payload: sectionIds[i]
+			})
+		}
+	})
+}
+
+/**
+ * Creates a new section and persist it on the backend, section is created with default values of addSection
+ *
+ * @param {int} position - positon of section
+ */
+export const createSection = (position) => async ({dispatch, state}) => {
+	const tempId = uuid()
+	const {lesson} = state
+	position = position || state.sections.length
+	dispatch(addSection({
+		_id: tempId,
+		position
+	}))
+	try{
+		const section = await editorWS.emit('create', `lesson/${lesson._id}/sections`, {})
+		dispatch({
+			type: REPLACE_ADDED_SECTION_ID,
+			payload: {
+				tempId,
+				backendId: section._id,
+			},
+		})
+	} catch (err){
+		dispatch({ // TODO: Recognice it and save it later, have to be a post and not a patch
+			// trigger could be the connection, need to be sockets as component und bind do a store
+			type: SAVING_SECTION_FAILED,
+			payload: {
+				_id: tempId
+			}
+		})
+	}
+}
+
+/**
+ * Subrutine to create diff of a serlo state and send it to backend
+ * 
+ * @param {string} lessonId - id of lesson,  the seciton belongs to
+ * @param {Object} section -
+ * 
+ * @returns {Promise} - Promise of the server request
+ */
+const createDiffAndSendToBackend = (lessonId, section) => {
+
+	const sectionDiff = buildDiff(
+		section.savedDocValue,
+		section.docValue,
+	)
+
+	return editorWS
+		.emit(
+			'patch',
+			`lesson/${lessonId}/sections/diff`,
+			section._id,
+			{state: sectionDiff},
+		)
+}
+
+const saveSectionDocValue = ({ignoreChangedSet = -1}) => async ({dispatch, state}) => {
+
+	dispatch({
+		type: SAVING_DOCVALUES
+	})
+
+	let proms = []
+	let sections = []
+	const hashes = []
+
+	if(ignoreChangedSet === -1){
```

Comment:
Die Verwendung von ignoreChangedSet um die Möglichkeit zu haben eine einzelne section zu triggern, müssen wir noch mal zusammen durchspechen. Das sieht sehr hacki aus.

Reply:
Wolle sagt fällt eh raus. -> in irgendein anderen Ticket, Cedric sagt: bitte dann im Ticket rein schreiben